runtime.md

```diff
@@ -15,3 +15,46 @@ Runs a process in a container. Can be invoked several times.
 Not sure we need that from oc cli. Process is killed from the outside.
 
 This event needs to be captured by oc to run onstop event handlers.
+
+## Hooks
+Hooks allow one to run code before/after various lifecycle events of the container. The state of the container is passed to the hooks over stdin, so the hooks could get the information they need to do their work.
+
+The path to the hooks could be absolute or just the name of the file. Hooks are executed from the host's filesystem. If the path to a hook isn't absolute then it is searched for within the PATH environment variable on the host.
```

Comment:
On Wed, Jul 22, 2015 at 04:26:14PM -0700, Mrunal Patel wrote:
> I think having paths as string handles Windows.

Ah, I was thinking that we might have fat containers that are
simultaneously compatible with several OSes.  For example:

  .
  ├── config.json
  └── rootfs
      ├── linux
      ├── os-x
      └── windows

But looking over the spec again, it seems like the separate Linux /
Windows / … sections are just for easy validation, and not a way to
setup things like:

  {
    …
    "linux": {
      "root": {
        "path": "rootfs/linux",
        …
      },
      …
    },
    "windows": {
      "root": {
        "path": "rootfs\\windows",
        …
      },
      …
    }
  }

Since that's not what we're shooting for, there's no need for per-OS
hook sections or cross-OS path-separator conventions.